topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/web/TopicOffsetRestController.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import uk.gov.gchq.palisade.service.topicoffset.model.Token;
+import uk.gov.gchq.palisade.service.topicoffset.model.TopicOffsetRequest;
+import uk.gov.gchq.palisade.service.topicoffset.stream.ConsumerTopicConfiguration;
+import uk.gov.gchq.palisade.service.topicoffset.stream.ProducerTopicConfiguration.Topic;
+
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * A REST interface mimicking the Kafka API to the service.
+ * POSTs to the controller write the request and headers to the upstream topic.
+ * These messages will then later be read by the service.
+ * Intended for debugging only.
+ */
+@RestController
+@RequestMapping(path = "/api")
+public class TopicOffsetRestController {
+    private final Sink<ProducerRecord<String, TopicOffsetRequest>, CompletionStage<Done>> upstreamSink;
+    private final ConsumerTopicConfiguration upstreamConfig;
+    private final Materializer materializer;
+
+    /**
+     * Autowired constructor for the rest controller
+     *
+     * @param upstreamSink   a sink to the upstream topic
+     * @param upstreamConfig the config for the topic (name, partitions, ...)
+     * @param materializer   the akka system materializer
+     */
+    public TopicOffsetRestController(
+            final Sink<ProducerRecord<String, TopicOffsetRequest>, CompletionStage<Done>> upstreamSink,
+            final ConsumerTopicConfiguration upstreamConfig,
+            final Materializer materializer) {
+        this.upstreamSink = upstreamSink;
+        this.upstreamConfig = upstreamConfig;
+        this.materializer = materializer;
+    }
+
+    /**
+     * REST endpoint for debugging the service, mimicking the Kafka API.
+     *
+     * @param headers a multi-value map of http request headers
+     * @param request the (optional) request itself
+     * @return the response from the service, or an error if one occurred
+     */
+    @PostMapping(value = "/offset", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<Void> topicOffset(
+            final @RequestHeader Map<String, String> headers,
+            final @RequestBody(required = false) TopicOffsetRequest request) {
+        // Process request as singleton list
+        this.topicOffsetMulti(headers, Collections.singletonList(request));
+
+        // Return result
+        return new ResponseEntity<>(HttpStatus.ACCEPTED);
+    }
```

Comment:
Do we want this put in a separate service per @dev-958 's suggestions or leave it for another time

Reply:
In a seperate service, like @pd104923 has done for the user service